doi/retro_doi_builder.py

```diff
@@ -0,0 +1,467 @@
+                dateString = match.group().replace("-", "")
+                date = datetime.strptime(dateString, "%Y%m%d")
+                epoch = datetime(1970, 1, 1)
+                return (date - epoch).total_seconds() * 1000
+
+            def getStreamedStop(start):
+                if start is None:
+                    return None
+                return start + MILLIS_PER_DAY - 1
+
+            conn.create_function("getStreamedStart", 2, getStreamedStart)
+            conn.create_function("getStreamedStop", 1, getStreamedStop)
+
+            conn.execute("DROP TABLE IF EXISTS PROVENANCE")
+            conn.execute(
+                    """CREATE TABLE PROVENANCE (subsite TEXT, node TEXT,
+                     sensor TEXT, method TEXT, deployment INTEGER,
+                     rowid TEXT, filename TEXT, parsername TEXT,
+                     parserversion TEXT)""")
+
+            for row in provenance:
+                subsite = row['subsite']
+                node = row['node']
+                sensor = row['sensor']
+                method = row['method']
+                deployment = row['deployment']
+                rowid = str(row['id'])
+                filename = row['fileName']
+                parsername = row['parserName']
+                parserversion = row['parserVersion']
+
+                record = [subsite, node, sensor, method, deployment, rowid,
+                          filename, parsername, parserversion]
+
+                conn.execute(
+                        """INSERT INTO PROVENANCE (subsite, node, sensor,
+                        method, deployment, rowid, filename, parsername,
+                        parserversion) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?);""",
+                        record)
+
+            conn.execute("""ALTER TABLE PROVENANCE add column
+                         start INTEGER""")
+            conn.execute("""UPDATE PROVENANCE SET
+                         start=getStreamedStart(method, filename)""")
+            conn.execute("""ALTER TABLE PROVENANCE add column
+                         stop INTEGER""")
+            conn.execute("""UPDATE PROVENANCE SET
+                         stop=getStreamedStop(start)""")
+
+            conn.commit()
+
+    def loadStreamMetadata(self):
+        with sqlite3.connect(self.database) as conn:
+            metadata = self.metadata_service_api.get_stream_metadata_records()
+            conn.execute("DROP TABLE IF EXISTS METADATA")
+            conn.execute(
+                    """CREATE TABLE METADATA (subsite TEXT, node TEXT,
+                     sensor TEXT, method TEXT, start INTEGER,
+                     stop INTEGER, stream TEXT)""")
+            for row in metadata:
+                subsite = row['referenceDesignator']['subsite']
+                node = row['referenceDesignator']['node']
+                sensor = row['referenceDesignator']['sensor']
+                method = row['method']
+                start = long(
```

Comment:
I'd rather see a call to a time function (e.g. `start = ntp_to_java_time(row['first'])`

repeats